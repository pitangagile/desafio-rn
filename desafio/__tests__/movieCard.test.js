import React from 'react';
import MovieCard from '../src/components/movieCard';

import renderer from 'react-test-renderer';
import { fireEvent, render } from '@testing-library/react-native';

describe('<MovieCard />', () => {
  it('MovieCard Snapshot', () => {
    const fakeData = [
      {
        title: 'Avatar',
        image:
          'https://http2.mlstatic.com/poster-cartaz-avatar-2-40x60cm-D_NQ_NP_701026-MLB25905406913_082017-O.webp',
      },
    ];
    const fragment = renderer
      .create(
        <MovieCard
          image={fakeData.image}
          title={fakeData.title}
          onPress={() => null}
        />,
      )
      .toJSON();
    expect(fragment).toMatchSnapshot();
  });

  it('MovieCard Press', () => {
    const fakeData = [
      {
        title: 'Avatar',
        image:
          'https://http2.mlstatic.com/poster-cartaz-avatar-2-40x60cm-D_NQ_NP_701026-MLB25905406913_082017-O.webp',
      },
    ];
    const pressTouch = jest.fn();

    const { getByTestId } = render(
      <MovieCard
        image={fakeData.image}
        title={fakeData.title}
        onPress={pressTouch}
      />,
    );

    fireEvent.press(getByTestId('touch'));
    expect(getByTestId('touch')).toBeTruthy();
    expect(pressTouch).toHaveBeenCalledTimes(0);
  });
});
