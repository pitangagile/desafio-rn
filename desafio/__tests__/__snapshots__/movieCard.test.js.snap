// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> MovieCard Snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "flexBasis": 0,
      "flexGrow": 1,
      "margin": 15,
      "marginBottom": 15,
      "opacity": 1,
      "width": "45%",
    }
  }
  testID="touch"
>
  <View
    style={
      Object {
        "borderRadius": 15,
        "height": 250,
        "marginBottom": 5,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": undefined,
        }
      }
      style={
        Object {
          "alignSelf": "flex-start",
          "flex": 1,
          "resizeMode": "stretch",
          "width": "100%",
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  />
</View>
`;
